backend/whatsapp.py

```diff
@@ -132,6 +136,10 @@ def onClose(self, ws):
         if self.onCloseCallback is not None and "func" in self.onCloseCallback:
             self.onCloseCallback["func"](self.onCloseCallback);
         eprint("WhatsApp backend Websocket closed.");
+    def keepAlive(self):
+        if self.activeWs is not None:
+            self.activeWs.send("?,,")
+            Timer(20.0, self.keepAlive).start()
```

Comment:
❓ Will this start a new timer each time `keepAlive` is called? That is, the number of active timers will continue to increase.

Reply:
every 20 seconds timer runs. not a lot of timers. `keepAlive` call it self after 20 sec.